heudiconv/heuristics/reproin.py

```diff
@@ -493,15 +522,30 @@ def infotodict(
         # Figure out the datatype_suffix (BIDS _suffix)
         #
         # If none was provided -- let's deduce it from the information we find:
-        # analyze s.protocol_name (series_id is based on it) for full name mapping etc
+        # analyze curr_seqinfo.protocol_name (series_id is based on it) for full name mapping etc
         if not datatype_suffix:
             if datatype == "func":
                 if "_pace_" in series_spec:
                     datatype_suffix = "pace"  # or should it be part of seq-
-                elif "P" in s.image_type:
-                    datatype_suffix = "phase"
-                elif "M" in s.image_type:
+                elif (
+                    "P" in curr_seqinfo.image_type
+                    and not curr_seqinfo.series_description.endswith("_SBRef")
+                ):
+                    datatype_suffix = "bold"
+                    series_info["part"] = "phase"
+                elif "M" in curr_seqinfo.image_type:
                     datatype_suffix = "bold"
+
+                    # if next one is phase fMRI, we should set part to mag
+                    if (
+                        (
+                            next_seqinfo.series_description
+                            == curr_seqinfo.series_description
+                        )
+                        and (next_dcm_image_iod_spec == "P")
+                        and not curr_seqinfo.series_description.endswith("_SBRef")
+                    ):
+                        series_info["part"] = "mag"
```

Comment:
Here's the main change. If necessary, I can try extending it to other suffixes (e.g., DWI).